drivers/adc/adc_ads114s0x.c

```diff
@@ -76,6 +76,26 @@ enum ads114s0x_register {
 	ADS114S0X_REGISTER_GPIOCON = 0x11,
 };
 
+/* available data rates described as samples per second, should be used with the time unit ticks */
+enum ads114s0x_data_rate {
+	ADS114S0X_CONFIG_DR_2_5 = 1, /* must not be equal to ADC_ACQ_TIME_DEFAULT */
```

Comment:
This definition is internal to the driver. How will the user know what value to use in the call to `ADC_ACQ_TIME()`? Wouldn't it be better to define such symbols in `dt-bindings/adc`?

Reply:
I absolutely agree with you, but I basically just stole the way to implement this from here:
https://github.com/zephyrproject-rtos/zephyr/blob/ea2aac6e5131bf916abd1e37334e94cc3481e1a8/drivers/adc/adc_ads1x1x.c#L54

Reply:
I just had a rather similar discussion in https://github.com/zephyrproject-rtos/zephyr/pull/56234#discussion_r1266990855 and my conclusion is that most drivers seem to expose this kind of information via a header. Therefore, I will go ahead and move this into a driver specific header.

Reply:
done

Reply:
I just realized you mentioned dt-bindings/adc, moved the enum definition there.